reporter/errors.go

```diff
@@ -39,31 +39,36 @@ type ErrorWithPos interface {
 
 // Error creates a new ErrorWithPos from the given error and source position.
 func Error(span ast.SourceSpan, err error) ErrorWithPos {
-	return errorWithSpan{SourceSpan: span, underlying: err}
+	var ewp ErrorWithPos
+	if errors.As(err, &ewp) {
+		// replace existing position with given one
+		return &errorWithSpan{SourceSpan: span, underlying: ewp.Unwrap()}
+	}
+	return &errorWithSpan{SourceSpan: span, underlying: err}
```

Comment:
I made some minor changes/improvements throughout this package, for better consistency.